BCStudentSoftwareDevTeam/celts
# Add recurring event issue46

Summary:

Added functionality to the recurring event checkbox. Where it displays an events table and the end datepicker. After the user selects an end date and start date the table will be populated with all of the events that would be created for that one recurring event.
Created a `calculateRecurringEventFrequency` function that checks the start date and end date and creates a list of dictionaries that contain the dates and event names for the new events that will be created.
Added an if conditional to the `createNewEvent` function that checks if the newly created event is supposed to be recurring or not and then either creates one event or multiple events for however many weeks the events recurs.
Wrote some test cases for  the `calculateRecurringEventFrequency` function and a test case for `createNewEvent`.

To Test:
Run the test on your terminal.
Go to the `<your_ip>:8080/<program_id>/create_event` page and then create an event as you normally would.
If you check the "this event occurs weekly" box then it should show the end date and the table.
Filling out the end date and start date should populate the table will the new events that will be created. 
Hitting save will create those events in the database.
You can double-check the events are created using PHPMyAdmin.

**NOTE:** This branch is forked from PR #37. This should probably not be merged until that one is.
